docs/source/configuration/volto-slate/api.md

```diff
@@ -0,0 +1,183 @@
+This allows a developer to modify and add several functionalities to a Slate `Editor`.
+For example, to define link elements as inline nodes, we override `isInline` from the `Editor` object.
+
+```js
+export const isInline = (editor) => {
+  const { isInline } = editor;
+
+  editor.isInline = (element) => {
+    return element && element.type === 'link' ? true : isInline(element);
+  };
+
+  return editor;
+};
+```
+
+
+(volto-slate-api-plugins-label)=
+
+## Plugins
+
+Plugins are the way to extend the capabilities of `volto-slate` by adding extra features to {term}`elementEditor`.
+
+See the [default plugins](https://github.com/plone/volto/tree/slate-integration/packages/volto-slate/src/editor/plugins).
+
+Refer to {ref}`writing-plugins-label` for how to write your own plugin.
+
+
+(volto-slate-api-SlateEditor-component-label)=
+
+## `SlateEditor` component
+
+`SlateEditor` is a top-level React component.
+It can be directly used in widgets to create a `slateJSON` field.
+For example, the following snippet creates a rich text `SlateEditor` component:
+
+```jsx
+<FormFieldWrapper {...props} draggable={false} className="slate_wysiwyg">
+  <div
+    className="slate_wysiwyg_box"
+    role="textbox"
+    tabIndex="-1"
+    onClick={() => {}}
+    onKeyDown={() => {}}
+  >
+    <SlateEditor
+      className={className}
+      readOnly={readOnly}
+      id={id}
+      name={id}
+      value={value}
+      onChange={(newValue) => {
+        onChange(id, newValue);
+      }}
+      block={block}
+      selected={selected}
+      properties={properties}
+      placeholder={placeholder}
+    />
+  </div>
+</FormFieldWrapper>
+```
+
+
+(volto-slate-api-elementEditor-label)=
+
+## `elementEditor`
+
+The {term}`elementEditor` is a top-level wrapper of all plugins used in `volto-slate` that exposes the plugins API in the form of `makeInlineElementPlugin`.
+It consists of various modules:
+
+`makeInlineElementPlugin`
+:   Used to build and install a custom schema based plugin from volto-slate API.
+    It expects a set of options passed as a property to your plugin.
+
+`PluginEditor`
+:   Editor component for your Plugin.
+
+`ToolbarButton`
+:   Custom plugin `ToolbarButton`.
+
+```{note}
+You will get to know more about `elementEditor` in {ref}`writing-plugins-label`.
+```
+
+
+(volto-slate-api-serialization-label)=
+
+## Serialization
+
+Serialization is the conversion of Slate JSON data into common formats such as text, HTML, and Markdown.
+Common serializers used in `volto-slate` are `serializeNodes`, `serializeNodesToText`, and `serializeNodesToHtml`.
+
+
+(volto-slate-api-deserialization-label)=
+
+## Deserialization
+
+Deserialization is the transformation of arbitrary inputs into a Slate-compatible JSON structure.
+`volto-slate` uses Slate's own mechanism of deserialization using `slate-hyperscript` package.
+Deserialization helps control what data comes into Slate per element.
+It is called before the editor's `Element` object for a plugin.
```

Comment:
The term `object` here is confusing. The deserialization methods are called before rendering slate Editor component, or `pluginEditor`, so the data is always in slateJson form.

Reply:
Would this be correct?

```suggestion
It is called before the `editorElement` component for a plugin.
```